### PR TITLE
Fix - Download AnalyzerManager async issue

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

When downloading AnalyzerManager, removing old version didn't wait, allowing the extract method to finish before deleting the old content resulting in removing the new analyzer manager as well.

This was the root caused of the following log that was shown:
```[DEBUG - 3:35:21 PM] Applicability runner could not find binary to run```

### CODE REVIEW

File: src/main/utils/resource.ts [hunk 1]
```diff
@@ -60,9 +60,9 @@ export class Resource {
      * if the given file is a binary, replace and copy it to the target path
      * @param tempPath - the file to copy into target
      */
-    public copyToTarget(tempPath: string) {
+    public async copyToTarget(tempPath: string) {
```
Comment (Or-Geva) [inline src/main/utils/resource.ts:63]: This function is async now, so to be more consistent, let's replace all other sync functions with async ones, such as copyFileSync -> copyFileSync

File: src/main/utils/resource.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export class Resource {
         let tmpFolder: string = ScanUtils.createTmpDir();
         try {
```
Comment (Or-Geva) [inline src/main/utils/resource.ts:83]: `ScanUtils.removeFolder(tmpFolder);` needs await?

File: src/main/utils/utils.ts [hunk 3]
```diff
@@ -151,9 +151,9 @@ export class Utils {
         }
     }
 
-    public static removeDirIfExists(dirPath: string) {
+    public static async removeDirIfExists(dirPath: string) {
         if (fs.existsSync(dirPath)) {
```
Comment (Or-Geva) [inline src/main/utils/utils.ts:155]: removeFolder already does the same as removeDirIfExists, we should remove code duplication here.

**Review by Or-Geva — CHANGES_REQUESTED** [review]

File: src/main/utils/utils.ts (hunk 3 above)
Comment (attiasas) [reply src/main/utils/utils.ts]: done

File: src/main/utils/resource.ts (hunk 2 above)
Comment (attiasas) [reply src/main/utils/resource.ts:83]: yes, `fse.remove` is async method this is why there was a bug before

**Pull request merged**